velox/exec/fuzzer/WindowFuzzer.h

```diff
@@ -0,0 +1,184 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include "velox/exec/Aggregate.h"
+#include "velox/exec/WindowFunction.h"
+#include "velox/exec/fuzzer/AggregationFuzzerBase.h"
+#include "velox/exec/fuzzer/PrestoQueryRunner.h"
+#include "velox/exec/fuzzer/ReferenceQueryRunner.h"
+#include "velox/vector/fuzzer/VectorFuzzer.h"
+
+DECLARE_bool(enable_window_reference_verification);
+
+namespace facebook::velox::exec::test {
+
+class WindowFuzzer : public AggregationFuzzerBase {
+ public:
+  WindowFuzzer(
+      AggregateFunctionSignatureMap aggregationSignatureMap,
+      WindowFunctionMap windowSignatureMap,
+      size_t seed,
+      const std::unordered_map<std::string, std::shared_ptr<ResultVerifier>>&
+          customVerificationFunctions,
+      const std::unordered_map<std::string, std::shared_ptr<InputGenerator>>&
+          customInputGenerators,
+      const std::unordered_set<std::string>& orderDependentFunctions,
+      VectorFuzzer::Options::TimestampPrecision timestampPrecision,
+      const std::unordered_map<std::string, std::string>& queryConfigs,
+      std::unique_ptr<ReferenceQueryRunner> referenceQueryRunner)
+      : AggregationFuzzerBase{seed, customVerificationFunctions, customInputGenerators, timestampPrecision, queryConfigs, std::move(referenceQueryRunner)},
+        orderDependentFunctions_{orderDependentFunctions} {
+    VELOX_CHECK(
+        !aggregationSignatureMap.empty() || !windowSignatureMap.empty(),
+        "No function signatures available.");
+
+    if (persistAndRunOnce_ && reproPersistPath_.empty()) {
+      std::cout
```

Comment:
nit: std::cerr ?

Reply:
This followed the code in AggregationFuzzer.cpp, but std::cerr does make more sense. I'll update it to std::cerr. Thank you for the suggestion!